src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/resource/EmailResource.java

```
@@ -0,0 +1,94 @@
+import org.fogbowcloud.saps.engine.core.archiver.storage.AccessLink;
+import org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorage;
+import org.fogbowcloud.saps.engine.core.archiver.storage.PermanentStorageType;
+import org.fogbowcloud.saps.engine.core.archiver.storage.exceptions.PermanentStorageException;
+import org.fogbowcloud.saps.engine.core.archiver.storage.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.archiver.storage.nfs.NfsPermanentStorage;
+import org.fogbowcloud.saps.engine.core.archiver.storage.swift.SwiftPermanentStorage;
+import org.fogbowcloud.saps.engine.core.dispatcher.email.TaskEmail;
+import org.fogbowcloud.saps.engine.core.dispatcher.email.TasksEmailSender;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.utils.SapsPropertiesConstants;
+import org.restlet.data.Form;
+import org.restlet.data.MediaType;
+import org.restlet.representation.Representation;
+import org.restlet.representation.StringRepresentation;
+import org.restlet.resource.Post;
+import org.restlet.resource.ResourceException;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+public class EmailResource extends BaseResource {
+
+    private static final Logger LOGGER = Logger.getLogger(EmailResource.class);
+
+    //TODO update value this to "tasks_id[]" (requires a change to the Dashboard component)
+    private static final String REQUEST_ATTR_PROCESSED_TASKS = "images_id[]";
+
+    @Post
+    public Representation sendTaskToEmail(Representation representation) {
+        Form form = new Form(representation);
+
+        String userEmail = form.getFirstValue(UserResource.REQUEST_ATTR_USER_EMAIL, true);
+        String userPass = form.getFirstValue(UserResource.REQUEST_ATTR_USERPASS, true);
+
+        if (!authenticateUser(userEmail, userPass) || userEmail.equals("anonymous"))
```

I think that authenticateUser should throw an exception itself once the authentication process hasn't worked...

---

I added FIXME about this.